tools/integration_tests/log_rotation/logrotate_logfile_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log_rotation
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"path"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/operations"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
+)
+
+const (
+	MiB          = 1024 * 1024
+	filePerms    = 0644
+	testFileName = "foo"
+)
+
+////////////////////////////////////////////////////////////////////////
+// Helpers
+////////////////////////////////////////////////////////////////////////
+
+func generateRandomData(t *testing.T, sizeInBytes int64) []byte {
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	data := make([]byte, sizeInBytes)
+	_, err := r.Read(data)
+	if err != nil {
+		t.Errorf("error while generating random data to write to file.")
+	}
+	return data
+}
+
+func runOperationsOnFileTillLogRotation(t *testing.T, wg *sync.WaitGroup, fileName string) {
+	defer wg.Done()
+
+	testDirPath := path.Join(setup.MntDir(), testDirName)
+	// Generate random data to write to file.
+	randomData := generateRandomData(t, MiB)
+	// Setup file to write to.
+	filePath := path.Join(testDirPath, fileName)
+	fh := operations.CreateFile(filePath, filePerms, t)
+
+	// Keep performing operations in mounted directory until log file is rotated.
+	var lastLogFileSize int64 = 0
+	for {
+		operations.WriteAt(string(randomData), 0, fh, t)
+
+		fi, err := operations.StatFile(logFilePath)
```

Comment:
I am not sure, t.Errorf will work inside go routine or not. Have we checked this by generating the test error.
If not then we can use ErrGroup instead of WaitGroup.

Reply:
I've verified that t.Errorf is working with waitgroup.